www/officers/coi.cgi

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env ruby
+PAGETITLE = "Conflict of Interest Affirmations"
+$LOAD_PATH.unshift '/srv/whimsy/lib'
+require 'wunderbar'
+require 'wunderbar/bootstrap'
+require 'whimsy/asf'
+require 'mail'
+require 'date'
+require 'tmpdir'
+
+coi_url = ASF::SVN.svnurl('conflict-of-interest')
+YEAR = DateTime.now.strftime "%Y"
+COI_CURRENT_URL = File.join(coi_url, YEAR)
+COI_CURRENT_TEMPLATE_URL = File.join(COI_CURRENT_URL, 'template.txt')
+
+user = ASF::Person.find($USER)
+USERID = user.id
+USERNAME = user.cn
+committees = ASF::Committee.officers + ASF::Committee.nonpmcs
+chairs = committees.map do |committee|
+ committee.chairs.map {|chair| chair[:id]}
+end
+IDS = (chairs.flatten + ASF::Service['board'].members.map(&:id)).uniq
+
+# Get the list of files in this year's directory
+signerfileslist, err = ASF::SVN.svn('list', COI_CURRENT_URL, {user: $USER, password: $PASSWORD})
+signerfiles = signerfileslist.split('\n')
+
+# Create the hash of {signer: signerurl} and remember user's affirmation file
+SIGNERS = Hash.new
+user_affirmation_file = nil
+signerfiles.each do |signerfile|
+  stem = signerfile[0..signerfile.index(".")-1]
+  user_affirmation_file = signerfile if stem == USERID
+  SIGNERS[stem] = signerfile unless stem == 'template'
+end
+USER_AFFIRMATION_FILE = user_affirmation_file
+
+# Get the list of required users who have not yet signed
+NONSIGNERS = []
+IDS.each do |required|
+  NONSIGNERS.push(required) unless SIGNERS.include? required
+end
+
+# Determine if user should sign the affirmation form
+user_is_required = IDS.include? USERID
+not_required_message = user_is_required ?' required':' not required'
+user_affirmation = SIGNERS.include? USERID
+have_affirmed_message = user_affirmation ? ' have affirmed' : ' have not affirmed'
+USER_IS_REQUIRED_BUT_NOT_AFFIRMED = (user_is_required and not user_affirmation)
+current_timestamp = DateTime.now.strftime "%Y-%m-%d %H:%M:%S"
+PANEL_MESSAGE = USER_IS_REQUIRED_BUT_NOT_AFFIRMED ?
+  'Sign Your Conflict of Interest Affirmation':
+  'Thank you for signing the Conflict of Interest Affirmation'
+
+# Read the template and append the signature block
+def get_affirmed_template(user, password, name, timestamp)
+  signature_block =
+  '       I, the undersigned, acknowledge that I have received,
+         read and understood the Conflict of Interest policy;
+         I agree to comply with the policy;
+         I understand that ASF is charitable and in order to maintain
+         its federal tax exemption it must engage primarily in activities
+         which accomplish one or more of its tax-exempt purposes.
+       Signed: __
+       Date: __
+       Metadata: __________Whimsy www/officers/coi.cgi___________'
+  template, err =
+    ASF::SVN.svn('cat', COI_CURRENT_TEMPLATE_URL, {user: user, password: password})
+  centered_name = "#{name}".center(50, '_')
+  centered_date ="#{timestamp}".center(51, '_')
```

Comment:
What if the name is longer than 50 characters?
AFAICT there is at least one such committer.

Reply:
I've changed it to 60 characters but if greater than that, the name line will be longer and not centered, which should be fine.